### DISCUSSION TITLE
Unsupported grant type with data-importer

### DISCUSSION THREAD

**Discussion by ProbablyBread** [firefly-iii/firefly-iii#6220] [Firefly III]
Hello, 

I'm trying to set up data-importer and I'm running into an issue with the OAuth authentication.

Screenshots: 
![oauth config](https://user-images.githubusercontent.com/50959479/177687837-419edc3d-147f-402f-8640-030b072cd629.png)
![oauth](https://user-images.githubusercontent.com/50959479/177687831-fdeb55ef-09fe-47e9-a045-62c10652156d.png)
![importer](https://user-images.githubusercontent.com/50959479/177687834-59528da6-0ef9-4bbf-aeb9-10229d9892b5.png)
![error](https://user-images.githubusercontent.com/50959479/177687833-8911c7e2-b176-4ad6-a909-36e5d44488f1.png)

Not too sure if the .env file for data-importer would be useful in troubleshooting, but here's the env file: 
[env.txt](https://github.com/firefly-iii/firefly-iii/files/9060826/env.txt)

I didn't touch anything inside except for TZ and reverse proxy. Since I'm using a self-signed certificate, I have also tried setting `VERIFY_TLS_SECURITY` to `false` but no luck. 

Here's the debug info for both firefly-iii and data-importer: 
Debug information generated at 2022-07-07 14:03:20 Asia/Singapore for Firefly III version **5.7.9**.

| Scope | Version |
| --- | --- |
| Firefly III | 5.7.9 |
| Firefly III API | 1.5.6 |
| Build | (unknown), (unknown) |
| DB version | 18 (exp. 18) |
| PHP | `8.0.20` |
| Host | `Linux` |

| System info | Value |
| --- | --- |
| System TZ | Asia/Singapore |
| Browser TZ | Asia/Singapore |
| App environment | local |
| App debug mode | false |
| App cache driver | file |
| App logging | notice, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Interface | fpm-fcgi |
| Default language | en_US |
| Default locale | equal |
| BCscale | 24 |
| DB drivers | mysql |
| Current driver | mysql |
| Trusted proxies (.env) | `**` |

| User info | Value |
| --- | --- |
| Login provider | eloquent  |
| User guard | web  |
| Headers | N/A, N/A  |
| Stateful domains |  |
| Session start | 2022-07-01 00:00:00 |
| Session end | 2022-07-31 23:59:59 |
| Session first | 2022-01-01 00:00:00 |
| User ID | 2 |
| User language | en_GB |
| User locale | en_GB |
| Attempt at "en_GB.utf8" | 'en_GB.utf8' |
| Attempt at "en_GB.UTF-8" | 'en_GB.UTF-8' |
| User agent | Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:102.0) Gecko/20100101 Firefox/102.0 |

Debug information generated at 2022-07-07 14:03:43 Asia/Singapore for Firefly III Data Importer version **0.9.9**.

| Scope | Version |
| --- | --- |
| FIDI | 0.9.9 |
| Build | (unknown), (unknown) |
| PHP | 8.0.20 |
| Host | Linux |

| System info | Value |
| --- | --- |
| Using docker? | false |
| System TZ | Asia/Singapore |
| Browser TZ | Asia/Singapore |
| App environment | local |
| App debug mode | false |
| App cache driver | file |
| App logging | debug, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Interface | fpm-fcgi |
| BCscale | 12 |
| Trusted proxies (.env) | ** |

| User info | Value |
| --- | --- |
| User agent | Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:102.0) Gecko/20100101 Firefox/102.0 |

**Comment by JC5** [comment]
I'm sorry I haven't seen this one before. I can only suggest clearing your cookies and cache etc.
